File: flutter_phonebook/lib/models/person.dart
import 'dart:convert';

List<Person> personFromJson(String str) => List<Person>.from(json.decode(str).map((x) => Person.fromJson(x)));

String personToJson(List<Person> data) => json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class Person {
    Person({
        required this.personId,
        this.personName,
        this.personSurname,
        this.telephoneNumber,
        this.notes,
    });

    int personId;
    String? personName;
    String? personSurname;
    String? telephoneNumber;
    String? notes;

    factory Person.fromJson(Map<String, dynamic> json) => Person(
        personId: json["personId"],
        personName: json["personName"],
        personSurname: json["personSurname"],
        telephoneNumber: json["telephoneNumber"],
        notes: json["notes"] == null ? null : json["notes"],
    );

    Map<String, dynamic> toJson() => {
        "personId": personId,
        "personName": personName,
        "personSurname": personSurname,
        "telephoneNumber": telephoneNumber,
        "notes": notes == null ? null : notes,
    };
}

File: PhonebookApplication/PhonebookApplication.Entities/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace PhonebookApplication.Entities
{
    public class Person
    {
        [Key]
        public int PersonId { get; set; }
        [Required]
        public string PersonName { get; set; }
        public string PersonSurname { get; set; }
        public string TelephoneNumber { get; set; }
        [MaxLength(50)]
        public string Notes { get; set; }
    }
}
